migration_scripts/bulk_load/src/footprint.clj

```diff
@@ -692,20 +802,86 @@
                     ;; offloaded, and return `nil`.
                     (when (seq (:cols stat)) stat))))))
 
-        extract-kvs
-        (fn [table stats]
-          (->> stats
-               (map #(extract-kv table %))
-               (filter some?)
-               (into [])))
-
-        extracted
-        (into {} (for [[table stats] tables
-                       :let [extract (extract-kvs table stats)]
-                       :when (seq extract)]
-                   [table extract]))]
+        extracted (filter-map-table-stats extract-kv tables)]
 
     ;; Materialize `extracted` into a dictionary first, so `kv-size`
     ;; holds the estimated key-value store size (`for` produces a lazy
     ;; sequence).
     (assoc extracted :kv-store [{:self @kv-size}])))
+
+;; Abstract IDs and Addresses ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn abstracted-ids
+  "Simulate replacing 32B Addresses and Object IDs with an 8B key.
+
+  Requires creating a mapping from long IDs to short IDs"
+  [tables]
+  (let [object-ids
+        (get-in tables [:objects-snapshot 0 :tuples])
+
+        txr-distinct
+        (get-in tables [:tx-recipients 0 :cols :recipient :distinct])
+
+        txr-tuples
+        (get-in tables [:tx-recipients 0 :tuples])
+
+        ;; If the `distinct` stat is negative, its absolute value is
+        ;; interpreted as the denominator of a ratio.
+        addresses
+        (cond (not txr-distinct)        nil
+              (not (neg? txr-distinct)) txr-distinct
+              :else (* txr-tuples (/ (- txr-distinct))))
```

Comment:
Is this `txr-tuples / abs(txr-distinct)`? I don't quite understand what this is supposed to represent - an estimate of unique addresses? But this would give us how many txs occurred per recipient?